tests/lib/rules/arrow-spaces.js

```diff
@@ -0,0 +1,104 @@
+/**
+ * @fileoverview Tests for arrow-spaces
+ * @author Jxck
+ * @copyright 2015 Jxck. All rights reserved.
+ */
+"use strict";
+
+//------------------------------------------------------------------------------
+// Requirements
+//------------------------------------------------------------------------------
+//
+
+var eslint = require("../../../lib/eslint"),
+    ESLintTester = require("eslint-tester");
+
+//------------------------------------------------------------------------------
+// Tests
+//------------------------------------------------------------------------------
+
+var eslintTester = new ESLintTester(eslint);
+
+var TT = { before: true, after: true };
+var TF = { before: true, after: false };
+var FT = { before: false, after: true };
+var FF = { before: false, after: false };
+
+var MB = { message: "Missing space before =>" };
+var MA = { message: "Missing space after =>" };
+var UB = { message: "Unexpected space before =>" };
+var UA = { message: "Unexpected space after =>" };
+
+var valid = [
+    [ TT, "a => a" ],
+    [ TT, "() => {}" ],
+    [ TT, "(a) => {}" ],
+
+    [ FT, "a=> a" ],
+    [ FT, "()=> {}" ],
+    [ FT, "(a)=> {}" ],
+
+    [ TF, "a =>a" ],
+    [ TF, "() =>{}" ],
+    [ TF, "(a) =>{}" ],
+
+    [ FF, "a=>a" ],
+    [ FF, "()=>{}" ],
+    [ FF, "(a)=>{}" ],
+
+    [ {}, "a => a" ],
+    [ {}, "() => {}" ],
+    [ {}, "(a) => {}" ]
+].map(function(param) {
+    return {
+        args: [1, param[0]],
+        code: param[1],
+        ecmaFeatures: { arrowFunctions: true }
+    };
+});
+
+var invalid = [
```

Comment:
For stylistic rules, its always good to also test for the column and row numbers.
We have seen issue with that in the past.

For reference: https://github.com/eslint/eslint/blob/master/tests/lib/rules/no-space-before-semi.js